### ISSUE TITLE
Cannot compile nimble 9.1 on R 4.0.2 using Mac OS X 10.15.6, Xcode 11.6

### ISSUE THREAD

**Issue by MooersLab** [nimble-dev/nimble#1053]
After installing R6 and igraph successfully, I failed to compile the nimble 9.1 from the nimble.org.
I am using R installed via Macports. 
Xcode was updated on July 17th.
I re-installed the X-code command line tools after this upgrade.

I was able to install the rstan and McElreath's rethinking package in R without trouble.

 

```r
install.packages("nimble", repos = "http://r-nimble.org", type = "source")

Installing package into ‘/Users/blaine/Library/R/4.0/library’
(as ‘lib’ is unspecified)
trying URL 'http://r-nimble.org/src/contrib/nimble_0.9.1.tar.gz'
Content type 'application/x-gzip' length 1902835 bytes (1.8 MB)
==================================================
downloaded 1.8 MB

* installing *source* package ‘nimble’ ...
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether /usr/bin/clang++ -std=gnu++11 accepts -g... yes
working on eigen
checking how to run the C++ preprocessor... /usr/bin/clang++ -std=gnu++11 -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking Eigen/Dense usability... no
checking Eigen/Dense presence... no
checking for Eigen/Dense... no
Using the version of Eigen provided with NIMBLE
ENABLE_LIB=false
configure: creating ./config.status
config.status: creating inst/make/Makevars
config.status: creating inst/make/Makevars_lib
config.status: creating R/config.R
config.status: creating src/Makevars
removing debugging flags
/opt/local/Library/Frameworks/R.framework/Resources/bin/Rscript --vanilla ./customizeMakeconf.R
Building libnimble.a
/usr/bin/clang++ -std=gnu++11 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -DR_NO_REMAP -I"../include"  -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk   -include /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp -I '/Users/blaine/Library/R/4.0/library/StanHeaders/include' -I '/Users/blaine/Library/R/4.0/library/RcppEigen/include' -fPIC  -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64  -c RcppUtils.cpp -o RcppUtils.o
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:22:30: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using double_return_t = std::conditional_t<std::is_const<std::remove_ref...
                        ~~~~~^~~~~~~~~~~~~
                             conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:22:63: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:26:31: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using reverse_return_t = std::conditional_t<std::is_const<std::remove_re...
                         ~~~~~^~~~~~~~~~~~~
                              conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:26:64: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:31:28: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using vari_return_t = std::conditional_t<std::is_const<std::remove_refer...
                      ~~~~~^~~~~~~~~~~~~
                           conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:31:61: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:36:31: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using forward_return_t = std::conditional_t<std::is_const<std::remove_re...
                         ~~~~~^~~~~~~~~~~~~
                              conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:36:64: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:60:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:66:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<(!std::is_pointer<T>::value && !is_fvar<T>::value
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:73:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<is_fvar<T>::value, forward_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:79:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_arithmetic<T>::value, double_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:148:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../include/Eigen/Dense:1:
In file included from ../include/Eigen/Core:415:
In file included from ../include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:154:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<!std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
14 errors generated.
make: *** [RcppUtils.o] Error 1
rm -f Makeconf
** libs
/usr/bin/clang++ -std=gnu++11 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -DR_NO_REMAP -I../inst/include    -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include   -include /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp -I '/Users/blaine/Library/R/4.0/library/StanHeaders/include' -I '/Users/blaine/Library/R/4.0/library/RcppEigen/include' -fPIC  -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64  -c RcppUtils.cpp -o RcppUtils.o
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:22:30: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using double_return_t = std::conditional_t<std::is_const<std::remove_ref...
                        ~~~~~^~~~~~~~~~~~~
                             conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:22:63: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:26:31: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using reverse_return_t = std::conditional_t<std::is_const<std::remove_re...
                         ~~~~~^~~~~~~~~~~~~
                              conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:26:64: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:31:28: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using vari_return_t = std::conditional_t<std::is_const<std::remove_refer...
                      ~~~~~^~~~~~~~~~~~~
                           conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:31:61: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:36:31: error: 
      no template named 'conditional_t' in namespace 'std'; did you mean
      'conditional'?
using forward_return_t = std::conditional_t<std::is_const<std::remove_re...
                         ~~~~~^~~~~~~~~~~~~
                              conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:440:33: note: 
      'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:36:64: error: 
      no template named 'remove_reference_t' in namespace 'std'; did you mean
      'remove_reference'?
  ...= std::conditional_t<std::is_const<std::remove_reference_t<T>>::value,
                                        ~~~~~^~~~~~~~~~~~~~~~~~
                                             remove_reference
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1131:50: note: 
      'remove_reference' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_referenc...
                                                 ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:60:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:66:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<(!std::is_pointer<T>::value && !is_fvar<T>::value
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:73:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<is_fvar<T>::value, forward_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:79:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_arithmetic<T>::value, double_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:148:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
In file included from <built-in>:1:
In file included from /Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from ../inst/include/Eigen/Dense:1:
In file included from ../inst/include/Eigen/Core:415:
In file included from ../inst/include/Eigen/src/Core/MatrixBase.h:130:
/Users/blaine/Library/R/4.0/library/StanHeaders/include/stan/math/prim/mat/eigen_plugins.h:154:10: error: 
      no template named 'enable_if_t' in namespace 'std'; did you mean
      'enable_if'?
    std::enable_if_t<!std::is_pointer<T>::value, reverse_return_t<T>>
    ~~~~~^~~~~~~~~~~
         enable_if
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:451:63: note: 
      'enable_if' declared here
template <bool, class _Tp = void> struct _LIBCPP_TEMPLATE_VIS enable_if {};
                                                              ^
14 errors generated.
make: *** [RcppUtils.o] Error 1
ERROR: compilation failed for package ‘nimble’
* removing ‘/Users/blaine/Library/R/4.0/library/nimble’

The downloaded source packages are in
	‘/private/var/folders/mv/1w6nby494g5b3vjf661ktyfr0000gn/T/RtmpBiIj1q/downloaded_packages’
Warning message:
In install.packages("nimble", repos = "http://r-nimble.org", type = "source") :
  installation of package ‘nimble’ had non-zero exit status

```

**Comment by MooersLab** [comment]
The error message suggested an issue with the StanHeaders pacakge.
I went to the rstan github site.
I found on this stan [page](https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started) the suggestion to run the [r-macos-rtools](https://github.com/rmacoslib/r-macos-rtools/blob/master/README.md)
I downloaded and ran the installer which does several things including moving old configuration files to backup status.
I had R open; I closed it.
I started R again. 
I reran the above install code without errors. 
I was able to load the nimble library and run the first example correctly. 
Other Mac OS users with Catalina may need the above helpful.
